### DISCUSSION TITLE
CPU Overload

### DISCUSSION THREAD

**Discussion by easiwriter** [pointfreeco/swift-composable-architecture#3442] [General]
The fan on my laptop is running continuously when I work on my app. I am slowly modularising it to reduce the length of compilations. I am using a 2.6 GHz Intel machine with 32Gb of memory. Do others ha the same problem?

**Comment by easiwriter** [comment]
Further to the post above I have regretfully had to abandon my attempt to modularise my app because of endless inexplicable compilation errors involving 'failed to find module' on import statements, despite the fact that the module was there. I've felt like I'm chasing my tail all the time and it was driving me nuts. I have no idea why this has been happening. I've tried clean builds in Xcode and swift builds from the terminal. One import works for no reason then another fails. I suspect as mentioned above that the problem is my laptop not having enough oomph and this is causing Swift to break. Any other suggestions welcome.

**Reply by stephencelis** [reply]
Have you looked at Activity Monitor to see if there are any runaway `swift` or macro processes? Sometimes force-quitting them can help. We're not sure there's anything we can do at the library level, but we're open if anyone has ideas. It just seems that there are some Xcode and Swift bugs that need to be addressed. Hopefully with Swift 6 out the team may be able to focus on stability and macro performance.

**Reply by easiwriter** [reply]
Thanks. I can't see any problems with this. Anyway, I have just made another attempt with the modularisation which still wont' compile. Here is an example:

In the package I have:

    .target (
      name: "FolderFormFeature",
      dependencies: [
        "ViewModels",
        "Models",
        .product(name: "ComposableArchitecture", package: "swift-composable-architecture"),
                    ]
    ),
    .target (
      name: "FolderFormView",
      dependencies: [
        "FolderFormFeature",
        "ViewModels",
        "Models",
        .product(name: "ComposableArchitecture", package: "swift-composable-architecture"),
                    ]
    ),

FolderFormView won't compile because it can't find FolderForm. Here's the view code:

<img width="1137" alt="Screenshot 2024-10-16 at 12 08 43" src="https://github.com/user-attachments/assets/882fac33-30e5-4170-950a-05861ffd8ae9">

And here's the Feature:

<img width="894" alt="Screenshot 2024-10-16 at 12 10 53" src="https://github.com/user-attachments/assets/576a66f6-d391-453c-a478-ad01f83898bb">

I have no idea why this is failing to compile, but it is typical of the kind of error I am seeing.

I am using Xcode 16.0

**Reply by stephencelis** [reply]
Sadly I think the bugs you are encountering are pretty general issues and not related specifically to TCA. A repro might help us give you more hints on what to check in Xcode, etc., but I'd suggest checking the activity logs to see if any errors are there that aren't getting surfaced directly in Xcode's UI, perhaps in an SPM-related log, or in a build log. You could also try resetting your packages, delete your derived data folder, etc. I've had to reboot my entire computer to address intermittent Xcode issues in the past...

**Reply by easiwriter** [reply]
Thanks Stephen

I will research this some more tomorrow (I am at least 5 hours ahead of you). If I don't make any headway would you be prepared to look at the project if I send a copy? The unmodularised project compiles OK, as do the examples you supply, so something weird is going on. If it compiles for you then the problem must be with my system.

Best
Keith

**Reply by stephencelis** [reply]
I can give it a shot!

**Reply by easiwriter** [reply]
I've discovered that if I get Xcode to build the package it works ok, but if I ask it to compile the app afterwards (no clean) it throws compilation errors. Not only that but it seems to compile as if it was a clean build when the only code is the single app source which imports the library.

The project is too big to upload so I've added you as a collaborator to the GitHub repo (I think).

**Reply by frolo1** [reply]
This sounds like an access issue. Maybe it helps making your feature public? 

https://docs.swift.org/swift-book/documentation/the-swift-programming-language/accesscontrol/#Access-Levels-for-Frameworks
> When you develop a framework, mark the public-facing interface to that framework as open or public so that it can be viewed and accessed by other modules, such as an app that imports the framework. This public-facing interface is the application programming interface (or API) for the framework.
